h2o-core/src/main/java/water/rapids/Merge.java

```diff
@@ -153,7 +159,10 @@ public static Frame merge(final Frame leftFrame, final Frame riteFrame, final in
     }
 
     BigInteger rightS = BigInteger.valueOf(256L<<riteShift);
-    long leftMSBto = riteBase.add(rightS).subtract(BigInteger.ONE).subtract(leftBase).shiftRight(leftShift).longValue();
+    long leftMSBto = riteFrameEmpty?(leftFrameEmpty? rightS.subtract(BigInteger.ONE).shiftRight(leftShift).longValue():
```

Comment:
this is unreadable, please extract into method or just a line of if/else statements

Reply:
Expanded it to if/else statements.  Hope it is clear.  Thanks, Wendy